aries_cloudagent/protocols/present_proof/v2_0/handlers/pres_request_handler.py

```diff
@@ -74,32 +75,49 @@ async def handle(self, context: RequestContext, responder: BaseResponder):
 
         # If auto_present is enabled, respond immediately with presentation
         if pres_ex_record.auto_present:
-            indy_proof_request = context.message.attachment(V20PresFormat.Format.INDY)
-
-            try:
-                req_creds = await indy_proof_req_preview2indy_requested_creds(
-                    indy_proof_request,
-                    preview=None,
-                    holder=context.inject(IndyHolder),
+            proof_request = context.message
+            input_formats = proof_request.formats
+            for format in input_formats:
+                pres_exch_format = V20PresFormat.Format.get(format.format)
+                if pres_exch_format is V20PresFormat.Format.INDY:
+                    try:
+                        indy_proof_request = proof_request.attachment(
+                            V20PresFormat.Format.INDY
+                        )
+                        req_creds = await indy_proof_req_preview2indy_requested_creds(
+                            indy_proof_request,
+                            preview=None,
+                            holder=context.inject(IndyHolder),
+                        )
+                    except ValueError as err:
+                        self._logger.warning(f"{err}")
+                        return
+
+                    request_data = {"requested_credentials": req_creds}
+                    (pres_ex_record, pres_message) = await pres_manager.create_pres(
+                        pres_ex_record=pres_ex_record,
+                        request_data=request_data,
+                        comment=(
+                            "auto-presented for proof request nonce "
+                            f"{indy_proof_request['nonce']}"
+                        ),
+                    )
+                elif pres_exch_format is V20PresFormat.Format.DIF:
```

Comment:
I think we shouldn't have format specific code in here but rather handle it in the format specific handler if that's possible. Ideally adding a new format should require you to change a minimal amount of files

Reply:
Removed all format-specific logic from `present_proof/v2_0/handlers/pres_request_handler.py`. Updated `present_proof/v2_0/formats/indy/handler.py` and `present_proof/v2_0/formats/dif/handler.py` accordingly.